torchao/prototype/optim_8bit/adam.py

```diff
@@ -0,0 +1,200 @@
+import torch
+from torch import Tensor
+from torch.optim import Optimizer
+
+from .subclass import DTQ8bit
+
+
+class AdamDTQ8bit(Optimizer):
+    def __init__(
+        self,
+        params,
+        lr=1e-3,
+        betas=(0.9, 0.999),
+        eps=1e-8,
+        weight_decay=1e-2,
+        amsgrad=False,
+        block_size=2048,
```

Comment:
is this the default block size in bnb? This is quite big

Reply:
Yes. It's specified in the paper (https://arxiv.org/pdf/2110.02861 section 2.1) and hard-coded in code https://github.com/TimDettmers/bitsandbytes/blob/dada530149212d64d4b69534716202659ef37ec8/csrc/kernels.cu#L4056-L4058